pylossless/pipeline.py

```diff
@@ -622,8 +633,34 @@ def flag_epoch_ic_sd1(self, raw):
         # icsd_epoch_flags=padflags(raw, icsd_epoch_flags,1,'value',.5);
 
     def save(self, raw, bids_path):
-        derivatives_path = bids_path.copy().update(extension='.fif', suffix='ll', root=bids_path.root / 'derivatives' / 'pylossless', check=False)
-        mne_bids.write_raw_bids(raw, derivatives_path, overwrite=True, format='EDF', allow_preload=True) #  TODO: address derivatives support in MNE bids.
+        lossless_suffix = bids_path.suffix + '_ll'
+        lossless_root = bids_path.root / 'derivatives' / 'pylossless'
+        derivatives_path = bids_path.copy().update(suffix=lossless_suffix,
+                                                   root=lossless_root,
+                                                   check=False
+                                                   )
+        mne_bids.write_raw_bids(raw,
+                                derivatives_path,
+                                overwrite=True,
+                                format='EDF',
+                                allow_preload=True)
+                                #  TODO address derivatives support in MNE bids.
+
+        # Save ICAs
+        for this_ica, self_ica, in zip(['ica1', 'ica2'],
```

Comment:
Maybe we only need to save the final ICA.

Reply:
Can you remind me of the purpose of ICA1 vs ICA2? Actually, what about documenting this in the default config file? I cannot judge on this question without having a clear idea of its purpose, but I see no reason not to save it. It is a small file and if it is used in the preprocessing, then for reproducibility, it is better to save it.

Reply:
the **timecourses** from the ICA1 decomposition are used to flag periods of time that have outlying ICA activation, the same way the pipeline flags bad periods of time with the channel data.

ICA2 is run excluding these bad times (flagged after the prior ICA), and the IC label function is run on this decomposition. This is also the final decomposition used for QCR.